CRM/Member/Form/MembershipRenewal.php

```diff
@@ -408,32 +468,75 @@ public function buildQuickForm() {
         );
       }
     }
-    $this->addFormRule(array('CRM_Member_Form_MembershipRenewal', 'formRule'));
+    $this->addFormRule(array('CRM_Member_Form_MembershipRenewal', 'formRule'), $this);
     $this->addElement('checkbox', 'is_different_contribution_contact', ts('Record Payment from a Different Contact?'));
     $this->addSelect('soft_credit_type_id', array('entity' => 'contribution_soft'));
     $this->addEntityRef('soft_credit_contact_id', ts('Payment From'), array('create' => TRUE));
   }
 
+  private function buildQuickFormPriceSet($orgId) {
+    if ($this->_submitValues) {
+      if (!$this->_submitValues['price_set_id']) {
+        // user is 'submitting' and didn't select price set.  So no need to
+        // add price set fields for validation.
+        return;
+      }
+      else {
+        // build price set, since we'll need to validate it.
+      }
+    }
+    else {
+      // not submitting values.  Build price set.
+    }
+    // get all price sets in system that have members
+    $priceSets = CRM_Price_BAO_PriceSet::getAssoc(FALSE, 'CiviMember');
+    // and create UI element for it
+    $this->add('select', 'price_set_id', ts('Price set'),
+          $priceSets,
+          NULL, array('onchange' => "buildAmount( this.value );")
+    );
+
+    // pass the price set id of default (or selected) price set back to smarty
+    $this->assign("priceSetId", $this->_priceSetId);
+    // set default on price set to value determined in buildQuickForm
+    $this->set("priceSetId", $this->_priceSetId);
+    // add all elements that are part of price set (these will be hidden to start)
+    // but will correspond to "last price set used".
+    CRM_Price_BAO_PriceSet::buildPriceSet($this);
+
+    // _priceSetOnly is set as request param via ajax onChange of price set field
+    // we pass it along to smarty template, so that it can exit early.
+    $this->assign("priceSetOnly", $this->_priceSetOnly);
+
+    $this->assign('price_set_has_new_orgs', $this->getPriceSetHasNewOrgs());
```

Comment:
I believe this is now redundant as you used the renewal date as the join date. The function and the js that hands of this var should be done for too now